servers/navigation_server_2d.cpp

```diff
@@ -305,3 +307,170 @@ bool FORWARD_1_C(agent_is_map_changed, RID, p_agent, rid_to_rid);
 void FORWARD_4_C(agent_set_callback, RID, p_agent, Object *, p_receiver, StringName, p_method, Variant, p_udata, rid_to_rid, obj_to_obj, sn_to_sn, var_to_var);
 
 void FORWARD_1_C(free, RID, p_object, rid_to_rid);
+
+Ref<NavigationPathQueryResult2D> NavigationServer2D::query_path(const Ref<NavigationPathQueryParameters2D> &p_query_parameters) const {
+	gd::PathQueryResult _query_result = NavigationServer3D::get_singleton()->_query_path(p_query_parameters->get_parameters());
+
+	Vector<Vector2> path_vec2;
+
+	for (int i = 0; i < _query_result.path.size(); i++) {
+		path_vec2.push_back(Vector2(_query_result.path[i].x, _query_result.path[i].z));
+	}
+
+	Ref<NavigationPathQueryResult2D> query_result(memnew(NavigationPathQueryResult2D));
+
+	query_result->set_path(path_vec2);
+	query_result->set_path_length(_query_result.path_length);
+
+	return query_result;
+}
+
+void NavigationPathQueryParameters2D::set_pathfinding_algorithm(const NavigationPathQueryParameters2D::PathfindingAlgorithm p_pathfinding_algorithm) {
+	switch (p_pathfinding_algorithm) {
+		case PATHFINDING_ALGORITHM_ASTAR: {
+			parameters.pathfinding_algorithm = gd::PathfindingAlgorithm::PATHFINDING_ALGORITHM_ASTAR;
+		} break;
+		default: {
+			parameters.pathfinding_algorithm = gd::PathfindingAlgorithm::PATHFINDING_ALGORITHM_ASTAR;
+		} break;
+	}
+}
+
+NavigationPathQueryParameters2D::PathfindingAlgorithm NavigationPathQueryParameters2D::get_pathfinding_algorithm() const {
+	switch (parameters.pathfinding_algorithm) {
+		case gd::PathfindingAlgorithm::PATHFINDING_ALGORITHM_ASTAR:
+			return PATHFINDING_ALGORITHM_ASTAR;
+		default:
+			return PATHFINDING_ALGORITHM_ASTAR;
+	}
+}
+
+void NavigationPathQueryParameters2D::set_map(const RID &p_map) {
+	parameters.map = p_map;
+}
+
+const RID &NavigationPathQueryParameters2D::get_map() const {
+	return parameters.map;
+}
+
+void NavigationPathQueryParameters2D::set_start_position(const Vector2 &p_start_position) {
+	start_position_vec2 = p_start_position;
+	parameters.start_position = Vector3(p_start_position.x, 0.0, p_start_position.y);
+}
+
+const Vector2 &NavigationPathQueryParameters2D::get_start_position() const {
+	return start_position_vec2;
+}
+
+void NavigationPathQueryParameters2D::set_target_position(const Vector2 &p_target_position) {
+	target_position_vec2 = p_target_position;
+	parameters.target_position = Vector3(p_target_position.x, 0.0, p_target_position.y);
+}
+
+const Vector2 &NavigationPathQueryParameters2D::get_target_position() const {
+	return target_position_vec2;
+}
+
+void NavigationPathQueryParameters2D::set_navigation_layers(uint32_t p_navigation_layers) {
+	parameters.navigation_layers = p_navigation_layers;
+};
```

Comment:
Personally, if these are one liners, I'd put them in the header, just to make things less verbose. But it depends if you think you are likely to add to the function implementations later.

Reply:
I think at some point all the setters could require some dirty flags to trigger update signals. I also have to admit I simply prefer this to one liners for my codeditor usage.